pkg/querier/worker/frontend_processor.go

```diff
@@ -95,6 +95,7 @@ func (fp *frontendProcessor) process(c frontendv1pb.Frontend_ProcessClient) erro
 			// and cancel the query.  We don't actually handle queries in parallel
 			// here, as we're running in lock step with the server - each Recv is
 			// paired with a Send.
+
```

Comment:
There's a question about whether a
 ctx = DecodeHTTPHeadersForLogging(ctx, request.HttpRequest) 
line is needed here, similar to what the scheduler_processor has. I'm in the process of testing that question right now, so depending on the results, either this line will disappear or be replaced.

Reply:
Confirmed that if the scheduler is not running, decoding needs to occur here. Not sure why there isn't tracing code being implemented here, but there is tracing being done on the scheduler_processor, maybe someone who knows more could explain.